File: beginner/greedy_method/section_scheduling/c++/sectionScheduling.cpp
#include <algorithm>
using namespace std;

// ===== ↓↓↓input↓↓↓ ===== //
const int MAX_N = 100000;
const int N = 5;
const int S[MAX_N] = {1, 2, 4, 6, 8}; 
const int T[MAX_N] = {3, 5, 7, 9, 10};
// ===== ↑↑↑input↑↑↑ ===== //

// Array for sort
pair<int, int> itv[MAX_N];

int main(int argc, char *argv[]) {
    // strage
    for (int i = 0; i < N; i++) {
        itv[i].first = T[i];
        itv[i].second = S[i];
    }
    sort(itv, itv + N);

    // t is process end time
    int ans = 0, t = 0;
    for (int i = 0; i < N; i++) {
        if (t < itv[i].second) {
            ans++;
            t = itv[i].first;
        }
    }

    printf("%d\n", ans);
}

File: beginner/greedy_method/section_scheduling/go/sectionScheduling.go
package main

import (
	"fmt"
	"sort"
)

// ===== ↓↓↓input↓↓↓ ===== //
const (
	MAX_N int = 100000
	N     int = 5
)

var S = []int{1, 2, 4, 6, 8}
var T = []int{3, 5, 7, 9, 10}

// ===== ↑↑↑input↑↑↑ ===== //

type Pair struct {
	x, y int
}

func main() {
	itv := make([]Pair, N)
	sort.Ints(T)
	sort.Ints(S)

	for i := 0; i < N; i++ {
		itv[i].x = T[i]
		itv[i].y = S[i]
	}

	ans := 0
	t := 0
	for i := 0; i < N; i++ {
		if t < itv[i].y {
			ans++
			t = itv[i].x
		}
	}

	fmt.Println(ans)
}
